tmt/base.py

```diff
@@ -2453,11 +2476,171 @@ def export(self):
         return data
 
 
-class Link(tmt.utils.SerializableContainer):
-    """ Core attribute link parsing """
+@dataclasses.dataclass
+class LinkNeedle:
```

Comment:
How about the name? I tried to stay away from `Link` or `link` since it may be easily mistaken for something else in this area of code, I wanted to express this is something code is supposed to find, or match against a collection of links while avoiding `LinkMatch`, because `Match` we know from `re` is something we get when things **did** match, the corresponding object in `re` lands would be a pattern. And I'm used to calling the second argument of "find X" functions `needle` (the first one is, obviously, `haystack`...)

@psss @lukaszachy @thrix what do you think?

Reply:
`LinkNeedle` sounds good to me.

Reply:
Or `LinkPattern`, that would probably be just a little bit more straightforward.